src/Components/NewPost.js

```diff
@@ -0,0 +1,181 @@
+import { useState } from 'react'
+
+import { createNewPost, getAllPosts } from '../Features/postsSlice'
+
+import {
+	MdiFeather,
+	IcBaselineClose,
+	DashiconsFormatGallery,
+} from '../Icones'
+
+import { StyledNewPost } from '../Components'
+
+import { useSelector, useDispatch } from 'react-redux'
+
+import { randomImgAPI } from '../utils/api'
+
+function NewPost() {
+	const [modal, setModal] = useState(false)
+
+	//add img source later in backed db
+	const user = useSelector(state => state.auth.user)
+
+	const { dp, username } = user
+
+	const imgSrc = !!dp ? dp : randomImgAPI
+
+	console.log(imgSrc)
+
+	const initialState = {
+		postText: '',
+		file: null,
+	}
+	const [newPost, setNewPost] = useState(initialState)
+
+	const [preview, setPreview] = useState(null)
+
+	const [uploadfileType, setUploadFileType] = useState('')
+
+	const handleSubmitPost = () => {}
+
+	const dispatch = useDispatch()
+
+	const handlePost = e => {
+		setNewPost(prev => {
+			return {
+				...prev,
+				postText: e.target.value,
+			}
+		})
+	}
+
+	const handleFormSubmit = e => {
+		e.preventDefault()
+
+		let newPostData = new FormData()
+
+		newPostData.append('postText', newPost.postText)
+
+		newPostData.append('postImgs', newPost.file)
+
+		dispatch(createNewPost(newPostData))
+
+		setNewPost(initialState)
+
+		setModal(prev => !prev)
+
+		window.location.reload()
```

Comment:
query: why are you reloading here?

Reply:
Thanks for ponting this . This was  to readily update state and show them in view. I have updated this and used useEffect and props which works without reaload